### PR TITLE
Issue #1560: Implement ARRAY_SLICE

### LINKED ISSUE #1560: Implement ARRAY_SLICE
It looks like the expression binder binds slice notation to a function called `array_slice`, which has not been implemented.

### PR DESCRIPTION
**Pull request by hawkfish**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -74,6 +74,7 @@ m4/lt~obsolete.m4
 #==============================================================================#
 #m4/
 build/
+duckdb_benchmark_data/
```
Comment (hannesmuehleisen) [inline .gitignore:77]: This folder should not be checked in

File: test/sql/types/list/test_list_index.test [hunk 2]
```diff
@@ -51,12 +51,38 @@ SELECT LIST_VALUE(42)[0]
...
+
+query T
+SELECT a[-5:] FROM (VALUES (LIST_VALUE(42, 43, 44))) tbl(a)
+----
+NULL
+
+query T
+SELECT a[:5] FROM (VALUES (LIST_VALUE(42, 43, 44))) tbl(a)
+----
+NULL
```
Comment (hannesmuehleisen) [inline test/sql/types/list/test_list_index.test:88]: Would it be possible to have some more tests for lists? e.g. empty list, NULL list etc?

**Review by hannes — COMMENTED** [review]
Looks great, two minor comments from me

File: .gitignore (hunk 1 above)
Comment (hannesmuehleisen) [reply .gitignore:77]: Oh never mind its in .gitignore

File: test/sql/types/list/test_list_index.test (hunk 2 above)
Comment (hannesmuehleisen) [reply test/sql/types/list/test_list_index.test:88]: Also tests with non-constant lists/parameters

**Comment by Mytherin** [comment]
Thanks a lot for the PR! Looks great to me now. @hannesmuehleisen any further comments, or can this be merged?

**Comment by hawkfish** [comment]
Thanks for making me add these tests - it turned up three bugs!

I find your lack of tests disturbing

**Pull request merged**